models/DataObject/ClassDefinition/Data/Fieldcollections.php

```diff
@@ -522,10 +522,16 @@ public function getGetterCode($class)
         // getter, no inheritance here, that's the only difference
         $key = $this->getName();
 
+        if ($this instanceof DataObject\ClassDefinition\Data\TypeDeclarationSupportInterface && $this->getReturnTypeDeclaration()) {
+            $typeDeclaration = ': ' . $this->getReturnTypeDeclaration();
+        } else {
+            $typeDeclaration = '';
+        }
+
         $code = '/**' . "\n";
         $code .= '* @return ' . $this->getPhpdocReturnType() . "\n";
         $code .= '*/' . "\n";
-        $code .= 'public function get' . ucfirst($key) . '()' . "\n";
```

Comment:
So this method was not generating the return type and conflicted with the change in

https://github.com/pimcore/pimcore/blob/19c1a82cce6e35d7aa78171f66c7ae428f75f739/bundles/EcommerceFrameworkBundle/src/Model/AbstractFilterDefinition.php#L62